### DISCUSSION TITLE
Avoid duplicate actions in same React view

### DISCUSSION THREAD

**Discussion by matzeeable** [mobxjs/mobx#2523] [Q&A] [answered]
Hi community!

![image](https://user-images.githubusercontent.com/1008534/96362463-c3234080-112d-11eb-9301-9f17e372c320.png)

I have a web application with the following components:

- Tabs `react-router`
- Tab has an additional view **"4 / 10 done"** with the state of the checklist: This state is always visible and not depending on the active tab
- **Dashboard** tab content

**My issue**: Currently, the user visits the dashboard and the `fetchChecklist()` action is called twice due to the fact the tab-container and the checklist-container in the dashboard grid is requesting the same data. At the end two REST-Requests are executed.

Is there any functionality / option in `mobx` or `mobx-react(-lite)` available to configure this behavior? How do you handle this?`

Regards,
Matthew 🙂

**Comment by danielkcz** [comment, marked as answer]
Without any code to show we can hardly advise you anything, but the usual way would be to lift the state up or in your case the request. I don't think this has anything to do with MobX.
